File: ios/DeviceModule.m
//
//  DeviceModule.m
//  AlamiAdibTS
//
//  Created by Adib on 14/08/22.
//

#import <Foundation/Foundation.h>

#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(DeviceModule,NSObject)

RCT_EXTERN_METHOD(getDeviceName: (RCTResponseSenderBlock)callback)
RCT_EXTERN_METHOD(getDeviceId: (RCTResponseSenderBlock)callbak)

@end

File: ios/DeviceModule.swift
//
//  DeviceModule.swift
//  AlamiAdibTS
//
//  Created by Adib on 14/08/22.
//

import Foundation

@objc(DeviceModule)
class DeviceModule: NSObject{
  
  
  @objc
  func getDeviceId(_ callback:RCTResponseSenderBlock){
    if let uuid = UIDevice.current.identifierForVendor?.uuidString {
        callback([uuid])
    }
  }
  
  @objc
  func getDeviceName(_ callback:RCTResponseSenderBlock){
    let name: String = UIDevice.current.name
    callback([name])
  }

  
  @objc
  static func requiresMainQueueSetup() ->Bool{
    return true;
  }
  
}
